tests/Unit/Send/ChainTest.php

```diff
@@ -139,4 +138,37 @@ public function testProcessNotProcessed() {
 
 		$this->chain->process($account, $localMessage);
 	}
+
+	public function testProcessDontProcessErrorerdMessages() {
+		$mailAccount = new MailAccount();
+		$mailAccount->setSentMailboxId(1);
+		$mailAccount->setUserId('bob');
+		$account = new Account($mailAccount);
+		$localMessage = new LocalMessage();
+		$localMessage->setId(100);
+		$localMessage->setStatus(LocalMessage::STATUS_ERROR);
+		$expected = new LocalMessage();
+		$expected->setStatus(LocalMessage::STATUS_IMAP_SENT_MAILBOX_FAIL);
+		$expected->setId(100);
+
+		$this->sentMailboxHandler->expects(self::any())
+			->method('setNext')
+			->withConsecutive(
+				[$this->antiAbuseHandler],
+				[$this->sentMailboxHandler],
+				[$this->copySentMessageHandler],
+				[$this->flagRepliedMessageHandler],
+			);
```

Comment:
sentMailboxHandler::setNext is only called once, not consecutively. try to change self::any() to self::exactly(4) and you should see failing assertions